servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -552,4 +587,32 @@ public Completable closeAsyncGracefully() {
             return closeable.closeAsyncGracefully();
         }
     }
+
+    private static final class SdStatusCompletable extends Completable {
+        private static final AtomicReferenceFieldUpdater<SdStatusCompletable, CompletableSource.Processor>
+                processorUpdater = newUpdater(SdStatusCompletable.class, CompletableSource.Processor.class,
+                "processor");
+        private volatile CompletableSource.Processor processor = newCompletableProcessor();
+        private boolean seenError;
+
+        @Override
+        protected void handleSubscribe(final Subscriber subscriber) {
+            processor.subscribe(subscriber);
+        }
+
+        void nextError(final Throwable t) {
+            final CompletableSource.Processor newProcessor = newCompletableProcessor();
+            newProcessor.onError(t);
```

Comment:
Because we expect any `Throwable` here, we use it as a way to update the state and always deliver the **latest** observed `t` to subscribers.
Since this is not on the hot path, it's ok to use this approach vs duplicating support for multiple subscribers from `CompletableProcessor`.